packages/dgrid/src/Table.ts

```diff
@@ -54,10 +68,31 @@ export class Table extends Common {
         super.enter(domNode, element);
     }
 
+    guessWidth(columns, data) {
+        for (const column of columns) {
+            if (column.children) {
+                const sampleData = [];
+                for (let i = 0; i < Math.min(3, data.length); ++i) {
+                    sampleData.push(data[i][column.idx]);
+                }
+                this.guessWidth(column.children, sampleData);
+            } else {
+                column.width = data.reduce((prevVal: number, row) => {
+                    const cell = ("" + row[column.idx]).trim();
+                    return Math.max(prevVal, textWidth(cell));
+                }, textWidth("" + column.label));
```

Comment:
This should run faster:
`column.width = Math.max.apply(undefined, data.map(row=>textWidth(("" + row[column.idx]).trim())))`

Reply:
Also, if we want a fast and precise way to measure the text... 
```javascript
measureText(font,text){
    this.text_ctx = this.text_ctx ? this.text_ctx : document.createElement('canvas').getContext('2d');
    this.text_ctx.font = font;
    return this.text_ctx.measureText(text).width;
}
measureText("12px Arial","some text to measure");
```